docs/WritingTests.asciidoc

```diff
@@ -428,24 +428,21 @@ Children B, C, D can run and finish anytime, but A must run until all B, C, D fi
 
 === Test synchronization and locking API
 
-OpenQA provides locking server through lock API. To use lock API import +lockapi+ package (_use lockapi;_) in your test file.
-Lock API provides functions: +mutex_create+, +mutex_lock+, +mutex_unlock+, +mutex_wait+. Each of these functions take at least one parameter: name of the lock. Note that lock name can't contain "-" character.
-Locks are associated with caller's job - locks can't be unlocked by different job then the one who locked the lock.
+OpenQA provides a locking mechanism. To use it in your test files import the +lockapi+ package (_use lockapi;_). It provides the following functions: +mutex_create+, +mutex_lock+, +mutex_unlock+, +mutex_wait+
```

Comment:
Another thing which is missing here is something like : 
` Also you need to make sure that jobs which suppose to use locking API are tied together by PARALLEL_WITH variable , otherwise job will not be able to do anything with mutex created by other job`
Not insisting on exactly this sentence but I think it is important to know and actually not so obvious as the fact that you need import lockapi to be able to use it :)

Reply:
That only jobs inside the same dependency cluster share the same "mutex namespace" is already mentioned in a subsequent paragraph.

Reply:
I think keep looking at this documentation with developer eyes.  What is "dependency cluster" suppose to mean for the user ? how to create it ? Is usage of START_AFTER_TEST is also creating dependency cluster or not ? Answers to all this questions might be obvious for you but please keep in mind that might be not so obvious for others. 
I think that straight and "dumb" definition via PARALLEL_WITH variable is more clear

Reply:
The previous way to express it ("test group defined by their dependencies") is not more specific. As it might cause confusion with job groups (due to the word "group") I rephrased it.

But your suggestion is indeed more specific so I'll mention that only this is about parallel dependency clusters specified via `PARALLEL_WITH` (without looking at the code to verify that this is actually correct, it just makes most sense and I trust you here).